components/eamxx/src/physics/dp/dp_functions.hpp

```diff
@@ -193,8 +198,45 @@ struct Functions
     const uview_1d<Spack>& t_update,   // temperature [m/s]
     const uview_2d<Spack>& q_update);  // tracer [vary]
 
-  KOKKOS_FUNCTION
-  static void iop_setinitial(const Int& nelemd, const uview_1d<element_t>& elem);
+  //---------------------------------------------------------
+  // Purpose: Set initial values from IOP files (where available)
+  //   when running SCM or DP-CRM.
+  //----------------------------------------------------------
+  static void iop_setinitial(
+    // Input arguments
+    const Int& plev,                        // number of vertical levels
+    const Int& pcnst,                       // number of advected constituents including cloud water
+    const Int& nelemd,                      // number of elements per MPI task
+    const Int& np,                          // NP
+    const Int& nstep,                       // the timestep number
+    const bool& use_replay,                 // use e3sm generated forcing
+    const bool& dynproc,                    // Designation of a dynamics processor - AaronDonahue
+    const bool& have_t,                     // dataset contains t
+    const bool& have_q,                     // dataset contains q
+    const bool& have_ps,                    // dataset contains ps
+    const bool& have_u,                     // dataset contains u
+    const bool& have_v,                     // dataset contains v
+    const bool& have_numliq,                // dataset contains numliq
+    const bool& have_cldliq,                // dataset contains cldliq
+    const bool& have_numice,                // dataset contains numice
+    const bool& have_cldice,                // dataset contains cldice
+    const bool& scm_zero_non_iop_tracers,   // Ignore all tracers from initial conditions file, and default all tracers not specified in IOP to minimum value (usually zero)
+    const bool& is_first_restart_step,      // is first restart step
+    const uview_1d<const Scalar>& qmin,     // minimum permitted constituent concentration (kg/kg) (pcnst)
+    const uview_1d<const Spack>& uobs,      // actual u wind
+    const uview_1d<const Spack>& vobs,      // actual v wind
+    const uview_1d<const Spack>& numliqobs, // actual ???
+    const uview_1d<const Spack>& numiceobs, // actual ???
+    const uview_1d<const Spack>& cldliqobs, // actual ???
+    const uview_1d<const Spack>& cldiceobs, // actual ???
+    const Scalar& psobs,                    // ???
+    const uview_1d<const Scalar>& dx_short, // short length scale in km (nelemd)
+    // Input/Output arguments
+    Scalar& dyn_dx_size,                    // for use in doubly periodic CRM mode
+    tracer_t& tracers,                      // tracers
```

Comment:
We could potentially avoid using tracers and elements directly by taking the individual items used from those objects and just piling them in to the function signature.

It looks like we use:
elem.m_forcing.m_ft
elem.m_state.m_ps_v
elem.m_state.m_v
elem.m_derived.m_omega_p

And
tracers.Q

Reply:
I like the idea of just passing the specific variables we use rather than big structures with a bunch of stuff we never use. I guess so long as we're just passing pointers there isn't any real memory impact, but it has annoyed me in the past that tons of unused variables are floating around in certain EAMf90 procs.

On the other hand, using elem directly gathers at least some of these variables into collections, minimizing your pain.